### PR TITLE
Adds support and validations for open content in Ion Schema 2.0

### PR DESCRIPTION
**Pull request by popematt**
**Issue #, if available:**

#207 

**Description of changes:**

* Adds support/validation for all aspects of open content in Ion Schema 2.0.

**Related PRs in ion-schema, ion-schema-tests, ion-schema-schemas:**

Test cases:
https://github.com/amzn/ion-schema-tests/issues/39

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: ion-schema/src/main/kotlin/com/amazon/ionschema/internal/SchemaImpl_2_0.kt [hunk 1]
```diff
@@ -27,46 +27,60 @@ import com.amazon.ionschema.IonSchemaVersion
...
          * INITIALIZATION ONLY.  This enables type B to find its already-loaded
          * dependency type A.  After initialization, [types] is expected to
          * be treated as immutable as required by the Schema interface.
          */
-    private val types: MutableMap<String, Type>
+    private val types: MutableMap<String, Type>,
 ) : SchemaImpl {
 
+    /**
+     * Represents reserved words that the user is reserving for their own open content.
```
Comment (zslayton) [inline ion-schema/src/main/kotlin/com/amazon/ionschema/internal/SchemaImpl_2_0.kt:60]: Is there a difference between a "reserved word" and a "reserved field?" If not, I suggest using a single term for consistency.

Also consider adding a comment with a link to [the relevant section of the spec](https://amzn.github.io/ion-schema/docs/isl-2-0/spec#open-content).

**Review by zslayton — APPROVED** [review]
👍

File: ion-schema/src/main/kotlin/com/amazon/ionschema/internal/SchemaImpl_2_0.kt (hunk 1 above)
Comment (popematt) [reply ion-schema/src/main/kotlin/com/amazon/ionschema/internal/SchemaImpl_2_0.kt:60]: There is a difference, and I'm using it incorrectly here. Good catch!

Comment (popematt) [reply ion-schema/src/main/kotlin/com/amazon/ionschema/internal/SchemaImpl_2_0.kt:60]:
```suggestion
     * Represents the collection of symbols that are declared as user reserved fields for a schema.
     * See https://amzn.github.io/ion-schema/docs/isl-2-0/spec#open-content
```

**Pull request merged**